internal/pkg/cli/run_local.go

```diff
@@ -515,9 +515,31 @@ func (o *runLocalOpts) prepareTask(ctx context.Context) (orchestrator.Task, erro
 		task.Containers[name] = ctr
 	}
 
+	// TODO (Adi): Use this dependency order in orchestrator to start and stop containers.
+	containerDeps := o.containerDependencies(mft.Manifest())
+	for name, dep := range containerDeps {
+		ctr, ok := task.Containers[name]
+		if !ok {
+			return orchestrator.Task{}, fmt.Errorf("missing container: %q is listed as a dependency, which doesn't exist in the task", name)
+		}
+		ctr.IsEssential = dep.IsEssential
+		ctr.DependsOn = dep.DependsOn
+	}
+
 	return task, nil
 }
 
+func (o *runLocalOpts) containerDependencies(unmarshaledManifest interface{}) map[string]manifest.ContainerDependency {
+	type containerDependency interface {
+		ContainerDependencies() map[string]manifest.ContainerDependency
+	}
+	mf, ok := unmarshaledManifest.(containerDependency)
+	if ok {
+		return mf.ContainerDependencies()
+	}
+	return nil
+}
```

Comment:
Does this serve any purpose to be a method of `runLocalOpts`?

Reply:
Good catch we can move this to `cli/deploy/workload.go`